packages/core/src/amazonq/lsp/lspClient.ts

```diff
@@ -0,0 +1,219 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*!
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+import * as vscode from 'vscode'
+import * as path from 'path'
+import * as nls from 'vscode-nls'
+import * as cp from 'child_process'
+import * as crypto from 'crypto'
+import * as jose from 'jose'
+
+import { Disposable, ExtensionContext } from 'vscode'
+
+import { LanguageClient, LanguageClientOptions, ServerOptions, TransportKind } from 'vscode-languageclient'
+import { GetUsageRequestType, IndexRequestType, QueryRequestType, UpdateIndexRequestType, Usage } from './types'
+import { Writable } from 'stream'
+import { CodeWhispererSettings } from '../../codewhisperer/util/codewhispererSettings'
+import { getLogger } from '../../shared'
+
+const localize = nls.loadMessageBundle()
+
+const key = crypto.randomBytes(32)
+
+/**
+ * Sends a json payload to the language server, who is waiting to know what the encryption key is.
+ * Code reference: https://github.com/aws/language-servers/blob/7da212185a5da75a72ce49a1a7982983f438651a/client/vscode/src/credentialsActivation.ts#L77
+ */
+export function writeEncryptionInit(stream: Writable): void {
+    const request = {
+        version: '1.0',
+        mode: 'JWT',
+        key: key.toString('base64'),
+    }
+    stream.write(JSON.stringify(request))
+    stream.write('\n')
+}
+/**
+ * LspClient manages the API call between VS Code extension and LSP server
+ * It encryptes the payload of API call.
+ */
+export class LspClient {
+    static #instance: LspClient
```

Comment:
don't think we need this LspClient to be a singleton?
maybe we should just manage it within LspController

Reply:
it must be a singleton. There should be only one Q LSP client for this server.